app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt

```diff
@@ -0,0 +1,418 @@
+import com.appcoins.payments.methods.adyen.presentation.rememberAdyenCreditCardUIState
+import com.appcoins.payments.methods.adyen.repository.NoNetworkException
+import com.aptoide.android.aptoidegames.R
+import com.aptoide.android.aptoidegames.SupportActivity
+import com.aptoide.android.aptoidegames.analytics.presentation.withAnalytics
+import com.aptoide.android.aptoidegames.feature_payments.AppGamesPaymentBottomSheet
+import com.aptoide.android.aptoidegames.feature_payments.LandscapePaymentErrorView
+import com.aptoide.android.aptoidegames.feature_payments.LandscapePaymentsNoConnectionView
+import com.aptoide.android.aptoidegames.feature_payments.LoadingView
+import com.aptoide.android.aptoidegames.feature_payments.PortraitPaymentErrorView
+import com.aptoide.android.aptoidegames.feature_payments.PortraitPaymentsNoConnectionView
+import com.aptoide.android.aptoidegames.feature_payments.PurchaseInfoRow
+import com.aptoide.android.aptoidegames.feature_payments.SuccessView
+import com.aptoide.android.aptoidegames.feature_payments.getAdyenErrorDescription
+import com.aptoide.android.aptoidegames.feature_payments.getAdyenErrorMessage
+import com.aptoide.android.aptoidegames.feature_payments.presentation.AdyenCreditCardStateEffect
+import com.aptoide.android.aptoidegames.feature_payments.presentation.PaymentButtons
+import com.aptoide.android.aptoidegames.theme.AGTypography
+import com.aptoide.android.aptoidegames.theme.AptoideTheme
+import com.aptoide.android.aptoidegames.theme.Palette
+import kotlinx.coroutines.delay
+import kotlin.random.Random
+
+private const val CREDIT_CARD_PAYMENT_ID_ARG = "paymentMethodId"
+private const val IS_PRE_SELECTED = "isPreSelected"
+private const val CREDIT_CARD_ROUTE = "payments/creditCard"
+private const val CREDIT_CARD_FULL_ROUTE =
+  "$CREDIT_CARD_ROUTE?$CREDIT_CARD_PAYMENT_ID_ARG={$CREDIT_CARD_PAYMENT_ID_ARG}&$IS_PRE_SELECTED={$IS_PRE_SELECTED}"
+
+private val creditCardPaymentArguments = listOf(
+  navArgument(CREDIT_CARD_PAYMENT_ID_ARG) {
+    type = NavType.StringType
+    nullable = false
+  },
+  navArgument(IS_PRE_SELECTED) {
+    type = NavType.BoolType
+    defaultValue = false
+    nullable = false
+  }
+)
+
+fun buildCreditCardRoute(
+  paymentMethodId: String,
+  isPreSelected: Boolean = false,
+) =
+  "$CREDIT_CARD_ROUTE?$CREDIT_CARD_PAYMENT_ID_ARG=${paymentMethodId}&$IS_PRE_SELECTED=${isPreSelected}"
+
+fun creditCardPaymentScreen(onFinish: (Boolean) -> Unit) = ScreenData.withAnalytics(
+  route = CREDIT_CARD_FULL_ROUTE,
+  screenAnalyticsName = "CreditCard",
+  arguments = creditCardPaymentArguments
+) { args, _, popBackStack ->
+  val paymentMethodId = args?.getString(CREDIT_CARD_PAYMENT_ID_ARG)!!
+  val isPreSelected = args.getBoolean(IS_PRE_SELECTED)
+  BuildAdyenCreditCardScreen(
+    onFinish = onFinish,
+    popBackStack = popBackStack,
+    paymentMethodId = paymentMethodId,
+  )
+
+  BackHandler(enabled = isPreSelected) {
+    onFinish(false)
+  }
+}
+
+@PreviewLight
```

Comment:
Should use PreviewDark

Reply:
All payments are in light theme, why do we need to preview dark here?